daemon/pool.go

```diff
@@ -884,10 +932,45 @@ func (p *movingAverage) Scale(total, idle, load int) int {
 		sum += v
 	}
 
-	avg := int(float64(sum) / float64(len(p.loads)))
+	avg := int(math.Ceil(float64(sum) / float64(len(p.loads))))
 	return p.sub.Scale(total, idle, avg)
 }
 
+type targMovingAverage struct {
+	sub     ScalingPolicy // policy that we will average
+	targets []int
+	pos     int
```

Comment:
I suggest including a separate `sum` field, so that the average computation doesn't require an extra loop per sample:

```go
type movingAverage struct {
	samples []int
	window  int
	pos     int
	sum     int
}

func (m *movingAverage) AddSample(v int) float64 {
	if len(m.samples) < cap(m.samples) {
		m.sum += v
		m.samples = append(m.samples, v)
		m.pos++
	} else {
		next := (m.pos + 1) % cap(m.samples)
		m.sum = (m.sum + v) - m.samples[next]
		m.pos = next
	}
	return math.Ceil(float64(m.sum) / float64(len(m.samples)))
}
```

or words to that effect. I don't think it's a performance issue, but I think the code is actually a little simpler this way.